### ISSUE TITLE
Message selection highlight is visually distracting

### ISSUE THREAD

**Issue by nhojb** [leggett/simplify#318]
The new message selection highlight is visually distracting - and initially confusing (why is that message at the top always grey?).

For me this is a big regression compared to the clean message view in Simplfy v1

Suggest behaviour similar to macOS list views (e.g. Finder)

1. Initially there is no selection highlight
2. Using the up/down arrow keys will show the highlight for selected message
3. Highlight persists until user clears via the Escape key

<img width="1141" alt="Screenshot 2020-08-31 at 17 09 27" src="https://user-images.githubusercontent.com/312540/91741913-4fdd6380-ebad-11ea-82ca-a9e30d5975cd.png">

**Comment by nhojb** [comment]
Perhaps add an option to disable keyboard navigation / highlighting of mailbox messages? Not something I ever really use. Though I do use the standard Gmail keyboard shortcuts all the time (star, delete, move etc).

Or a less visually intrusive method of indication the selected message?

**Comment by leggett** [comment]
This is definitely one of the things I expected to be controversial that I left in to get feedback from a wider audience. 

Finder isn't quite comparable in my mind. 

At a high level, I think using keyboard shortcuts makes people more productive and as a result happier. So I am trying to encourage keyboard shortcut usage. 

But I concede that as implemented, it is probably too much for a wide audience.

**Comment by leggett** [comment]
I'd also ask that you live with it for a week and see how you feel then. I doubt it will change... and first impressions are important. But I am curious. Do you find yourself using keyboard shortcuts more at all? Is it still visually annoying or have you gotten used to it?

**Comment by jayarr100** [comment]
The first-message highlighting confused me at first as well, and I initially thought it was a bug.

After remembering your introductory email and then hitting the arrow keys, I quickly realized what was going on. And, as noted to you in an email, I _do_ find I'm using the keyboard shortcuts for message list navigation more routinely already. (I'm a heavy keyboard shortcut user in general, but I'd never managed to get myself in the habit of using the shortcuts for that function, specifically, before.)

The lack of an initial highlighting could be one way to solve it, although that might counteract the element's benefit and intended purpose. I wonder if an onboarding note — an overlay that appears at some point during early usage and explains why the first message is highlighted and how you can interact with it — along with an option to disable that system entirely, if wanted, might be a good way to improve it? 

Another thought: This made me wonder how that function behaved when Gmail keyboard shortcuts are disabled. I just checked, and the highlighting visual still appears — despite the fact that, in that scenario, you're unable to open messages with the enter key (thus making the highlighting somewhat irrelevant and confusing). In addition to or instead of a broader on/off switch, could that feature's presence be mapped to the keyboard shortcut function, so that it appears when those are enabled but does not appear when they're disabled?

**Comment by gchiam** [comment]
I also feel that the selection highlight is quite distracting, but less so under the dark theme.

**Comment by nhojb** [comment]
I'm not convinced I will find it any less distracting with time (though of course we learn to live with all sorts of UI, good and bad, eventually).

For me the message selection highlight shouts out "this message is important, look at ME". For example, I may have just switched to a tab with a few new messages, all of which I want to quickly visually scan, but the highlight grabs my attention and gets in my way (even when I know its purpose). I think this visual distraction is the reason that lists on macOS tend not to show any selection highlight until something is, you know, selected.

The reason I love Simplify is that it is not visually distracting, no single UI element is shouting for attention. The message highlight seems to go against this (wonderful) spirit.

**Comment by leggett** [comment]
Lots of good comments here. Really appreciate it. Not sure what change I will make but I'm pretty confident some change is needed.

**Comment by freste** [comment]
I use keyboard shortcuts - I quite like the highlight.
Some sort of indication of 'current' message is needed, and I already don't remember what it used to be....aha, a blue bar at left, w/bolded checkbox and star. That worked ok ; though now that I notice it, that blue bar is jarring isn't it.

**Comment by leggett** [comment]
In my mind, it is just very invisible. And in Gmail's sea of icons and color swatches, I just ignore it. 

I've started to think of ways I might be able to dim the effect until you use keyboard shortcuts (use a less strong background) or maybe it fades out after the page loads until you use shortcuts. I'm not sure... need to sleep on it.

**Comment by leggett** [comment]
I know I told y'all to sit with it for a while... but first impressions matter. So, I'm going to try a twist on @nhojb's original suggestion and make the highlight less bold initially. As soon as you use up or down, the highlight gets stronger. 

I'll leave this open for further discussion.

**Comment by jayarr100** [comment]
You know, I do prefer this, personally, in terms of ongoing usability — as that initial message highlight was slightly distracting to me (as others have also mentioned). And once I was aware of its existence, having that constant dark highlight (before I was even attempting to select messages) didn't really serve any positive purpose for me.

The one thing I notice now, though, is that in an inbox with a darker background, the _new_ initial message highlight is extraordinarily subtle, to the point of being almost unnoticeable — particularly when the messages involved are already read and thus not bolded:

![simplify-highlight](https://user-images.githubusercontent.com/40185140/91855253-8b6a4300-ec2a-11ea-919a-6d2d80652306.jpg)

This doesn't bother me, personally, since I already know the feature exists. But I do wonder whether it'd cause the feature to be entirely lost upon new users (in this specific, darker-background scenario), as they'd probably only realize it was there if they happened to press the arrow keys and shift in the darker-highlight mode.

**Comment by leggett** [comment]
Yeah, I probably went too light. I picked a color while using the default all white theme and didn't check other themes. I'll separate it out and apply a different color for different types of themes.

**Comment by ibmkahm** [comment]
i would also prefer if the message at the top  is not grey by default, only after selecting it

**Comment by nhojb** [comment]
FWIW I think the selection highlight in v2.1.9 is pretty much perfect (in "light" mode).

**Comment by ibmkahm** [comment]
> FWIW I think the selection highlight in v2.1.9 is pretty much perfect (in "light" mode).

even after updating the extension, mine is still at 2.1.8. I wonder what changed that you like it now

**Comment by leggett** [comment]
The extension isn't auto updating?? Maybe it takes a day or two but I definitely expect it to auto update. Will investigate.

**Comment by alecbz** [comment]
Responding from https://github.com/leggett/simplify/issues/363#issuecomment-686908814:

> toned this down until you use up/down. Curious how you feel about that change.

Hm, I think that's a bit better.

How do you feel about just disabling it until someone presses up/down? Like, for users who never use the up/down shortcuts, it feels like it's just visual noise that's not really communicating anything meaningful. For users who do use up/down, it seems not crazy to have the first use of up/down start showing the indicator?

**Comment by ibmkahm** [comment]
> How do you feel about just disabling it until someone presses up/down? Like, for users who never use the up/down shortcuts, it feels like it's just visual noise that's not really communicating anything meaningful. For users who do use up/down, it seems not crazy to have the first use of up/down start showing the indicator?

i would also prefer this!

**Comment by leggett** [comment]
I've updated the colors used in v2.1.11 for the dark theme (more muted) and any image theme (less muted) to make them more like the default theme. 

I'd like to avoid adding a setting for this. if I continue to get pushback, I will probably fade out the highlight after N seconds instead and bring it back on using any keyboard shortcuts. 

Thanks for all the feedback and brainstorming. Closing for now.

**Comment by nhojb** [comment]
One more thought :-)

I was using Gmail "virgin" the other day and was reminded of the default message selection highlight (vertical line at left edge of selected message). Although the v2.1.10 approach is ok, I think the default Gmail solution is (for once) a better one (perhaps with a more muted colour).

**Comment by jayarr100** [comment]
Sorry to complicate matters right after this resolution, but seeing your update made me realize something — so I thought I'd share:

I realize now that ever since Simplify moved to this more muted coloring (on a dark theme, which is what I use for my primary Gmail account), I've gotten back out of the habit of using keyboard shortcuts for selecting and opening messages in the inbox.

I'm just one person, of course, so take it for what it is. But I'm also a huge fan of keyboard shortcuts in general and someone who typically embraces them wholeheartedly. For whatever reason, I've just never managed to get myself in the habit of using keyboard shortcuts for navigating the message list in the inbox, specifically. During that initial period, when the more prominent highlighting was in place, I did. Then, when it went away and became significantly more muted, I stopped — without even realizing it until now.

This might lend some (admittedly limited) support to your original theory that the more prominent highlighting would encourage keyboard shortcut use and lead to a better overall experience. That's certainly been the case for me.

Maybe the initial-highlight-then-fade approach, especially in conjunction with a first-time-use tooltip of some sort, could be a good in-between solution that achieves the originally intended benefit without being overkill and risking annoyance or confusion.

**Comment by leggett** [comment]
No need to apologize. I always welcome feedback / thoughts. 

I haven't released v2.1.11, so it is set to get even _more_ muted on the dark theme soon. I will write this down as something to come back to before full release. There is definitely a difficult balance to strike... some people just aren't going to use keyboard shortcuts (which is fine) and this will forever be annoying to them.
